mlrun/api/schemas/constants.py

```diff
@@ -52,3 +52,30 @@ class HeaderNames:
     patch_mode = f"{headers_prefix}patch-mode"
     deletion_strategy = f"{headers_prefix}deletion-strategy"
     secret_store_token = f"{headers_prefix}secret-store-token"
+
+
+class GroupByField(str, Enum):
+    name = "name"  # Supported for feature-store objects
+    key = "key"  # Supported for artifacts
+
+    def to_group_by_db_field(self, db_cls):
+        if self.value == GroupByField.name:
+            return db_cls.name
+        else:
+            return db_cls.key
```

Comment:
I think we better have a per object enum class

Reply:
Well, currently `artifacts` code is not in place yet, so I see no point in defining the enum for `artifacts`. I will change this to be `FeatureStoreGroupByField` and remove the `key` option. I don't see much point in having a different enum for `FeatureSet` and for `FeatureStore` as a lot of the code treats them the same (which is part of the point of the way they're built anyway).